heron/tools/cli/src/python/execute.py

```diff
@@ -101,8 +101,15 @@ def heron_tar(class_name, topology_tar, arguments, tmpdir_root, java_defines):
   '''
   # Extract tar to a tmp folder.
   tmpdir = tempfile.mkdtemp(dir=tmpdir_root, prefix='tmp')
+  tmpdir_abs = os.path.abspath(tmpdir)
 
   with contextlib.closing(tarfile.open(topology_tar)) as tar:
+    # Validate file paths
+    for name in tar.getnames():
+      untarpath_abs = os.path.abspath(os.path.join(tmpdir, name))
+      if not untarpath_abs.startswith(tmpdir_abs):
+        raise Exception(f"tar attempted to extract a file outside the destination directory: {name}")
```

Comment:
maybe useful to log `untarpath_abs` as well.

Reply:
```suggestion
        raise Exception(f"tar attempted to extract a file outside the destination dir: {untarpath_abs}:{name}")
```